### PR TITLE
chore(progressbar): rename and move spinner-determinate to progressbar-rounded #3963

### LINKED ISSUE #3963: Consider Spinner Determinate to be a progress element
There is an element [Spinner Determinate](https://baseweb.design/components/spinner/#spinner-determinate) and it lives under the **Spinner** section.
But from the semantic point, doesn't it act as a progress bar? What do you think about moving this component to the **Progress bar** page, as an "Oval (Circle) Progress", for example?

### PR DESCRIPTION
**Pull request by Deliaz**
Fixes #3963 

#### Description
I am proposing to rename `SpinnerDeterminate` into `ProgressBarRounded`.
Please consider the code changes. 

This PR includes the following actions: 
1. renamed component
2. moved related files to the `progressbar` folder
3. moved doc example to the Progress Bar page
4. kept re-exports in `Spinner` for backward compatibility

Unfortunately, there is no way to hide `StyledSpinnerDeterminate..` exports from the Spinner page. 
However, I propose to plan the full removal of SpinnerDeterminate exports, as a planned deprecation. 

#### Scope
Minor: New Feature

### CODE REVIEW

**Comment by Deliaz** [comment]
Conflict resolved. PR updated.

**Comment by tajo** [comment]
This is a breaking change?

**Comment by Deliaz** [comment]
@tajo 
It shouldn't be. 
I've tried to save the same exports, here: 
https://github.com/uber/baseweb/pull/3968/files#diff-60e96902f6b01794fda40a2d58a77bcbd650658dd082d8e2e9ab95cbcbbdf87bR29-R34

**Comment by tajo** [comment]
lgtm

**Review by tajo — APPROVED** [review]

**Comment by tajo** [comment]
You should merge the VRT PR into yours and also update component sizes.

**Comment by Deliaz** [comment]
I did merge VRT. What is the update needs for component sizes?

**Comment by tajo** [comment]
> I did merge VRT. What is the update needs for component sizes?

One of the failing CI steps. Running this [script](https://github.com/uber/baseweb/blob/master/scripts/component-sizes.js) checks size of each component and compare it to the current values (json file). You have to update it since this PR naturally increased the size of progressbar significantly.

**Comment by Deliaz** [comment]
@tajo thanks, the `component-sizes.json` has been updated.

**Comment by tajo** [comment]
Ah, it seems that the visual snapshots are flaky because of the animation? It could be disabled or having some fixed value instead of changing over time.

**Comment by Deliaz** [comment]
Ah, right. That's why I am constantly updating it :D
Do you want me to update test cases?

**Comment by tajo** [comment]
> Ah, right. That's why I am constantly updating it :D
> Do you want me to update test cases?

That would be great. I'm not sure why it wasn't flaky before. You didn't make really any changes to tests, right?

**Comment by Deliaz** [comment]
Nope, I just moved them between folders and renamed the description.

File: src/spinner/index.js [hunk 1]
```diff
@@ -5,20 +5,31 @@ This source code is licensed under the MIT license found in the
...
 // @flow
+import {
+  StyledProgressBarRoundedRoot,
+  StyledProgressBarRoundedSvg,
+  StyledProgressBarRoundedText,
+  StyledProgressBarRoundedTrackBackground,
+  StyledProgressBarRoundedTrackForeground,
+} from '../progress-bar/index.js';
 export {default as Spinner} from './spinner.js';
-export {default as SpinnerDeterminate} from './spinner-determinate.js';
```
Comment (tajo) [inline src/spinner/index.js:9]: Ah, I missed this but it seems that this main export is now missing.

```jsx
import {SpinnerDeterminate} from 'baseui/spinner';
<SpinnerDeterminate progress={progress} />
```

doesn't work anymore.

Comment (Deliaz) [reply src/spinner/index.js:9]: Ah, right! 
Updated.

**Comment by tajo** [comment]
Thanks!

**Pull request merged**